packages/server/graphql/public/mutations/createStripeSubscription.ts

```diff
@@ -38,14 +38,18 @@ const createStripeSubscription: MutationResolvers['createStripeSubscription'] =
     return standardError(new Error('Organization already has a subscription'), {userId: viewerId})
   }
   const {email} = viewer
-  const customer = stripeId
-    ? await manager.retrieveCustomer(stripeId)
-    : await manager.createCustomer(orgId, email)
-  const {id: customerId} = customer
-  const res = await manager.attachPaymentToCustomer(customerId, paymentMethodId)
-  if (res instanceof Error) return standardError(res, {userId: viewerId})
-  // wait until the payment is attached to the customer before updating the default payment method
-  await manager.updateDefaultPaymentMethod(customerId, paymentMethodId)
+  let customer: Stripe.Response<Stripe.Customer | Stripe.DeletedCustomer>
+  if (stripeId) {
+    customer = await manager.retrieveCustomer(stripeId)
+    const {id: customerId} = customer
+    const res = await manager.attachPaymentToCustomer(customerId, paymentMethodId)
+    if (res instanceof Error) return standardError(res, {userId: viewerId})
+    // wait until the payment is attached to the customer before updating the default payment method
+    await manager.updateDefaultPaymentMethod(customerId, paymentMethodId)
```

Comment:
For the edge case where a user is upgrading for a second time, we could use a `Promise.all` to speed things up a bit here?  Something like:

```
  const [attachPaymentResult, updateDefaultPaymentResult] = await Promise.all([
    manager.attachPaymentToCustomer(customerId, paymentMethodId),
    manager.updateDefaultPaymentMethod(customerId, paymentMethodId)
  ])
  if (attachPaymentResult instanceof Error) return standardError(attachPaymentResult, {userId: viewerId})
```

Reply:
i tried this...
nope, gotta attach first. ah well, not too bad since re-upgrade is such an edge case!